### ISSUE TITLE
Transparent proxy mode

### ISSUE THREAD

**Issue by nihaals** [ProxymanApp/Proxyman#1701]
## Description

It would be great if there was a way to proxy programs that don't use the system proxy.

## Why this feature/change is important?

Proxyman currently has [automatic setup](https://docs.proxyman.io/automatic-setup/automatic-setup) which covers common CLI cases but doesn't cover everything. A transparent proxy mode would be able to cover the remaining cases.

See:
- #236
- Burp Suite's [invisible proxy mode](https://portswigger.net/burp/documentation/desktop/tools/proxy/invisible) ([older guide](https://portswigger.net/support/using-burp-suites-invisible-proxy-settings-to-test-a-non-proxy-aware-thick-client-application))
- mitmproxy's [transparent proxy docs](https://docs.mitmproxy.org/stable/howto-transparent/)
- Surge's [enhanced mode](https://manual.nssurge.com/others/enhanced-mode.html)

**Comment by NghiaTranUIT** [comment]
Thanks @nihaals I've tried but I hit the wall:

- mitmproxy transparent proxy mode: Unfortunately, it's required to open the `mitmproxy` with noby user (sudo -u nobody mitmproxy), which is impossible to achieve on Proxyman app. => Apple doesn't allow to start macOS app with this mode.
- Surge's Enhanced Mode: I don't have much knowledge to create a virtual network interface (VIF). 

I will spend more time to research and find a better solution.

**Comment by nihaals** [comment]
I don't know about the trade-offs, but my experience with Surge's approach was positive and would make the toggle/launch experience as smooth as it is currently, but I also don't really know how it works.

**Comment by marcelocecin** [comment]
https://www.txthinking.com/talks/articles/mobile-capture-en.article

**Comment by NghiaTranUIT** [comment]
Thanks @marcelocecin for the link. Unfortunately, it's the same approach that we're currently using for Proxyman iOS (By using a Network Extension): https://apps.apple.com/us/app/proxyman-network-debug-tool/id1551292695

For the macOS, Transparent Mode with Virtual Network interface is completely different, and I'm not sure how to tackle it : /

**Comment by jtg** [comment]
Would really like this feature. Some apps are not proxy aware and are quite a pain to try and get through proxyman. I am currently using a workaround by using Burp's transparent proxy mode. This requires many steps to make DNS work correctly, but could all be done in proxyman. Burp requires no permissions, but features may be more limited than the others - I'm not sure what the difference is. Here is what I'm doing.

Enable Burp in transparent proxy mode on port 443.

Edit /etc/hosts to have the domains that the target app is reaching (example: 127.0.0.1     api.mytestapp.com)

Configure Burp to use an upstream proxy. Set this upstream proxy to the proxyman server and port. Burp would normally then allow interception, but otherwise transparently redirect any incoming requests based on the Host: header for non-proxy aware apps, to the correct destination. However, to avoid this, you also configure Burp itself to use an upstream http proxy. This essentially rewrites the non-proxy aware app's request to be proxy aware before passing to proxyman.

Configure proxyman to use DNS Spoofing. This isn't actually to spoof the DNS, but to UNSPOOF what you did with /etc/hosts. Otherwise, when proxyman attempts to send it back out, it would also be fooled by /etc/hosts, and send it back to localhost on 443 where Burp is listening.

I would have used Burp for the whole thing, but proxyman offers options that Burp doesn't have, such as response mapping. Proxyman could be updated to include the same functionality that Burp offers and that could be skipped. Still unfortunate that pre-defining every host in /etc/hosts is required, but better than nothing.

**Comment by marcelocecin** [comment]
perhaps this could be achievable with the help of an LLM specializing in programming

**Comment by njivaRandrznt** [comment]
I’d really love to see this feature in Proxyman. It looks like some other apps, like fluxzy and now fiddler everywhere, already support it without additional dependencies, but I’d much prefer to have it available directly in Proxyman.

**Comment by SpacyRicochet** [comment]
Chiming in here, in case it helps anyone. I've managed to get Burp working with by adding the Intercept53 extension, following this [blog post](https://blog.snoopbees.com/why-your-proxy-doesnt-work-on-flutter-apps-and-how-to-fix-it-bba9d7c9b3f2).

I'm also looking forward to seeing something like this working in Proxyman, which is so much nicer than Burp. That would definitely make me renew my license 😄

**Comment by NghiaTranUIT** [comment]
@SpacyRicochet your blog is about capturing HTTPS from a Flutter app. If it's the Flutter, Proxyman has a simple solution: https://docs.proxyman.com/debug-devices/flutter

No need to install the Intercept53 Extension with a lot of manual configs

**Comment by SpacyRicochet** [comment]
> @SpacyRicochet your blog is about capturing HTTPS from a Flutter app. If it's the Flutter, Proxyman has a simple solution: https://docs.proxyman.com/debug-devices/flutter
> 
> No need to install the Intercept53 Extension with a lot of manual configs

Thanks for the reply! 

That sounds useful, though it gives some questions. I'm inspecting the traffic not of my own apps, but of public Play Store apps. 

* Can you install Play Store apps or downloaded apks on the Android emulators? I'm not an Android developer, so I have no experience with this.
* This doesn't address iOS apps that are installed from the App Store either, which the blog post does cover.

My current 'ideal' scenario would Proxyman having a 'Capture Flutter/Dart traffic' button, that:
* Elevates Proxyman's process to root (or provides instructions with the terminal command to do this),
* Does the capturing and routing at the DNS level like Burp + Intercept53,
* Provides the instruction about setting the DNS provider on the device _instead of_ setting the Proxy.

**Comment by NghiaTranUIT** [comment]
@SpacyRicochet 

> Can you install Play Store apps or downloaded apks on the Android emulators? I'm not an Android developer, so I have no experience with this.

Yes, the latest version of Proxyman macOS, supports Magisk, which helps you to root the Google Play Store Emulator. It means you can download the Flutter app on PlayStore on this Emulator and Proxyman can capture it.

> This doesn't address iOS apps that are installed from the App Store either, which the blog post does cover.

Yes, iOS Flutter is different. The only solution I know is using Proxyman iOS: https://apps.apple.com/us/app/proxyman-capture-https/id1551292695

Proxyman iOS has a built-in VPN, which can capture HTTP/HTTPS on your iPhone.

> My current 'ideal' scenario would Proxyman having a 'Capture Flutter/Dart traffic' button, that:
Elevates Proxyman's process to root (or provides instructions with the terminal command to do this),
> Does the capturing and routing at the DNS level like Burp + Intercept53,

No. Not yet.

> Provides the instruction about setting the DNS provider on the device instead of setting the Proxy.

Proxyman has the Transparent Proxy, which can works like the DNS tool
